python/services/dataservice/dmod/dataservice/service.py

```diff
@@ -221,77 +290,85 @@ def _process_dataset_create(self, message: DatasetManagementMessage) -> DatasetM
 
         # Create the dataset
         dataset_type = self._determine_dataset_type(message)
-        self._all_data_managers[dataset_type].create(name=message.dataset_name, category=message.data_category,
-                                                     domain=message.data_domain, is_read_only=False)
-        return DatasetManagementResponse(success=True, reason="Dataset Created", is_awaiting=message.is_pending_data)
-
-    async def can_be_fulfilled(self, requirement: DataRequirement) -> Tuple[bool, Optional[str]]:
+        dataset = self._all_data_managers[dataset_type].create(name=message.dataset_name, category=message.data_category,
+                                                               domain=message.data_domain, is_read_only=False)
+        # TODO: determine if there is an expectation to find data
+        # TODO:     if so, attempt to find data, setting pending response based on result
+        return DatasetManagementResponse(action=message.management_action, success=True, reason="Dataset Created",
+                                         data_id=str(dataset.uuid), dataset_name=dataset.name,
+                                         is_awaiting=message.is_pending_data)
+
+    async def can_be_fulfilled(self, requirements: List[DataRequirement]) -> Tuple[bool, Optional[str]]:
         """
-        Determine whether this requirement for this job can be fulfilled, either directly or by deriving a new dataset.
+        Determine whether all the given requirements can be fulfilled, either directly or by deriving a new dataset.
 
-        The returned tuple will return two items.  The first is whether the data requirement can be fulfilled given the
+        The returned tuple will return two items.  The first is whether the data requirements can be fulfilled given the
         currently existing datasets.  The second is the name of the fulfilling dataset, if a fulfilling dataset already
-        exists.  If data among known datasets is sufficient to fulfill the requirement, but deriving a new dataset is
+        exists.  If data among known datasets is sufficient to fulfill the requirements, but deriving a new dataset is
         necessary (e.g., in a different format, or by combining data from multiple datasets), then the second value will
         be ``None``.
 
         Parameters
         ----------
-        requirement : DataRequirement
-            The data requirement in question that needs to be fulfilled.
+        requirements : List[DataRequirement]
+            The data requirements in question that need to be fulfilled.
 
         Returns
         -------
         Tuple[bool, Optional[str]]
-            A tuple of whether the requirement can be fulfilled and, if one exists, the name of the fulfilling dataset.
+            A tuple of whether the requirements can be fulfilled and, if one exists, the name of the fulfilling dataset.
```

Comment:
Does this need to return a tuple PER requirement now?

Reply:
Let's ignore this one for now.  I went back and forth on whether this should be one requirement or several (in this function and several others).  It eventually got cleaned up, but the merge process got messy, so I don't want to play too much with it.